src/crimson/osd/replicated_backend.cc

```diff
@@ -142,3 +144,35 @@ seastar::future<> ReplicatedBackend::stop()
   pending_trans.clear();
   return seastar::now();
 }
+
+seastar::future<bool>
+ReplicatedBackend::already_complete(const osd_reqid_t& reqid,
+				    const eversion_t& at_version)
+{
+  auto iter = pending_trans.begin();
+  if (iter == pending_trans.end()) {
+    return seastar::make_ready_future<bool>(true);
+  }
+  auto& pending_txn = iter->second;
+  if (pending_txn.at_version > at_version) {
```

Comment:
i don't really follow. what this implies is

> if the first pending_txn's `at_version` is greater than the `at_version` of the request we 
> are about to replay, we can consider the request committed already.

right? can we assume that if *any* pending_txn whose `at_version` is greater than `at_version` of this request, this request is committed?